nncf/utils.py

```diff
@@ -358,3 +359,23 @@ def training_mode_switcher(model: torch.nn.Module, is_training: bool = True):
         yield
     finally:
         model.train(is_original_mode_training)
+
+
+def compute_FLOPs_hook(module, input_, output, dict_to_save, name):
+    if isinstance(module, (nn.Conv1d, nn.ConvTranspose1d)):
+        ks = module.weight.data.shape
+        mac_count = ks[0] * ks[1] * ks[2] * output.shape[2]
+    elif isinstance(module, (nn.Conv2d, nn.ConvTranspose2d)):
+        ks = module.weight.data.shape
+        mac_count = ks[0] * ks[1] * ks[2] * ks[3] * output.shape[2] * output.shape[3]
+    elif isinstance(module, (nn.Conv3d, nn.ConvTranspose3d)):
+        ks = module.weight.data.shape
+        mac_count = ks[0] * ks[1] * ks[2] * ks[3] * ks[4] * output.shape[2] * \
+                    output.shape[3] * output.shape[4]
+    elif isinstance(module, nn.Linear):
+        mac_count = input_[0].shape[1] * output.shape[-1]
+    elif isinstance(module, (nn.BatchNorm1d, nn.BatchNorm2d, nn.BatchNorm3d)):
```

Comment:
We should not count BatchNorms because they are fused at inference time.

Reply:
BatchNom is not taking into account in FLOPs calculation anymore.